tools/lxdclient/client_network.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// +build go1.3
+
+package lxdclient
+
+import (
+	"fmt"
+
+	"github.com/lxc/lxd"
+	"github.com/lxc/lxd/shared"
+	"github.com/juju/errors"
+
+	"github.com/juju/juju/network"
+)
+
+type rawNetworkClient interface {
+	NetworkCreate(name string, config map[string]string) error
+	NetworkGet(name string) (shared.NetworkConfig, error)
+}
+
+type networkClient struct {
+	raw       rawNetworkClient
+	supported bool
+}
+
+// create a network
+func (c *networkClient) NetworkCreate(name string, config map[string]string) error {
+	if !c.supported {
+		return errors.NotSupportedf("network API not supported on this remote")
+	}
+
+	return c.raw.NetworkCreate(name, config)
+}
+
+// acquire a network's configuration
+func (c *networkClient) NetworkGet(name string) (shared.NetworkConfig, error) {
+	if !c.supported {
+		return shared.NetworkConfig{}, errors.NotSupportedf("network API not supported on this remote")
+	}
+
+	return c.raw.NetworkGet(name)
+}
+
+type creator interface {
+	rawNetworkClient
+	ProfileDeviceAdd(profile, devname, devtype string, props []string) (*lxd.Response, error)
+	ProfileConfig(profile string) (*shared.ProfileConfig, error)
+}
+
+// Create a default bridge and (if necessary) insert it into the default profile.
+func CreateDefaultBridgeInDefaultProfile(client creator) error {
```

Comment:
@stgraber wrote:
"The n.Type check is only used in our client for when someone does "lxc network attach eth0 blah" where eth0 isn't a bridge."

This implies that your client (I assume lxc) can do the bridge set up and that very similar code has been put into Juju.

This is all a bit academic though because unless LXD grows an API call to set up the bridge that both Juju and LXC can call, then we need to duplicate the code.